crates/noirc_evaluator/src/ssa_refactor/ssa_builder/mod.rs

```diff
@@ -29,8 +29,14 @@ use super::{
 /// Contrary to the name, this struct has the capacity to build as many
 /// functions as needed, although it is limited to one function at a time.
 pub(crate) struct FunctionBuilder {
+    /// The current function being built.
+    /// A function builder can be re-used to build different functions.
+    ///
+    /// Note: What is the benefit of this vs creating a function builder per function?
```

Comment:
I'm not sure actually. This was taken from cranelift which does reuse contexts and clears some internal Vecs in-between so that it does not allocate more often by creating new Vecs each time. We don't share any Vecs though so the most benefit we get from it is a slightly nicer API I think with `builder.finish()` not relying on users to accumulate the function Vec themselves and pass it in.